appcues/src/main/java/com/appcues/ui/ExperienceRenderer.kt

```diff
@@ -55,6 +55,8 @@ internal class ExperienceRenderer(override val scope: AppcuesScope) : AppcuesCom
     private val potentiallyRenderableExperiences = hashMapOf<RenderContext, List<Experience>>()
     private val previewExperiences = hashMapOf<RenderContext, Experience>()
 
+    private val modalStateMachineOwner = ModalStateMachineOwner(get(::onExperienceEnded), get())
```

Comment:
so instead of extracting the modalStateMachineOwner from the stateMachines object, would it make more sense to have some logic around the stateMachiens.cleanup() that could call "stopRetry"? maybe each StatemachineOwner can have their on cleanup() method, so things that are particular to each StateMachineOwner implementation can be handled internally